site/documentation/content/api/telemetry/index.md

```diff
@@ -118,3 +121,134 @@ The *Business Application* can only consume telemetry messages that have been up
+| all   | 0  | >0 | `false` | Setting *acks* to `all` allows for adapters to implement both *AT LEAST ONCE* or *AT MOST ONCE* delivery semantics, depending on whether the adapter waits for and considers the acknowledgement it receives from the Kafka cluster or not. Disabling retries might cause messages to fail in case of high load or transient transmission failures. |
+| all   | >0 | 1  | `false` | Setting *acks* to `all` allows for adapters to implement both *AT LEAST ONCE* or *AT MOST ONCE* delivery semantics, depending on whether the adapter waits for and considers the acknowledgement it receives from the Kafka cluster or not. When retries are enabled, *max.in.flight.requests.per.connection* MUST be set to `1`. |
+| all   | >0 | <6 | `true`  | Setting *acks* to `all` allows for adapters to implement both *AT LEAST ONCE* or *AT MOST ONCE* delivery semantics, depending on whether the adapter waits for and considers the acknowledgement it receives from the Kafka cluster or not. |
+
+All other combinations are not supported by Hono.
+{{% /note %}}
+
+The recommended Kafka producer properties for *AT LEAST ONCE* delivery are: `enable.idempotence=true`, `acks=all`, `max.in.flight.requests.per.connection=5`, leaving *retries* unset, and setting *delivery.timeout.ms* to a reasonable value (supported from Kafka version 1.0.0 on). 
+
+The following sequence diagram illustrates the flow of messages involved in the *HTTP Adapter* producing a telemetry data message to the Kafka cluster implementing *AT MOST ONCE* delivery semantics.
+
+{{< figure src="produce_kafka_qos0.svg" title="Produce telemetry data flow (AT MOST ONCE)" >}}
+
+1. *Device* `4711` PUTs telemetry data to the *HTTP Adapter*
+   1. *HTTP Adapter* produces telemetry data to *Kafka Cluster*.
+   1. *HTTP Adapter* acknowledges the reception of the data to the *Device*.
+
+The following sequence diagram illustrates the flow of messages involved in the *HTTP Adapter* producing a telemetry data message to the Kafka cluster implementing *AT LEAST ONCE* delivery semantics.
+
+{{< figure src="produce_kafka_qos1.svg" title="Produce telemetry data flow (AT LEAST ONCE)" >}}
+
+1. *Device* `4711` PUTs telemetry data to the *HTTP Adapter*, indicating *QoS Level* 1.
+   1. *HTTP Adapter* produces telemetry data to *Kafka Cluster*.
+   1. *Kafka Cluster* acknowledges reception of the message.
+   1. *HTTP Adapter* acknowledges the reception of the data to the *Device*.
+
+When the Kafka producer receives an exception when sending a telemetry message to Kafka, the protocol adapter MUST NOT try to re-send such rejected messages but SHOULD indicate the failed transfer to the device if the transport protocol provides means to do so.
+
+**Message Format**
+
+A Kafka message (also called record) consists of a key, a value, a timestamp, and headers.
+The key of the message MUST be the device ID.
```

Comment:
which device ID is that? The authenticated device's or the device's that the data originates from?

Reply:
The ID of the device that the data originates from. I'm going to add this.